### ISSUE TITLE
feed_through_consumed_energy is total_increasing but negative

### ISSUE THREAD

**Issue by gdgib** [SpanPanel/span#11]
```
2023-04-12 18:26:42.646 WARNING (Recorder) [homeassistant.components.sensor.recorder] Entity sensor.feed_through_consumed_energy from integration span_panel has state class total_increasing, but its state is negative. Triggered by state -54967.208492206 with last_updated set to 2023-04-13T01:19:59.004244+00:00. Please report it to the custom integration author.
```

Honestly not sure whether this is due to solar causing negative energy consumption, or maybe just a silly sign error (e.g. a negative number needs to be inverted before passing it to HA).

**Comment by etmelvin** [comment]
Honesty, I don’t really know what feedthrough means and have never really tried to use it for anything; I mainly stick with the Main Meter entities for my energy dashboard. I’ve seen this behavior on the previous integration versions too. 

It’ll be nice if we have some documentations on these exposed properties from the API, maybe we can ask the SPAN employees on the forum?

**Comment by gdgib** [comment]
Agreed, that makes sense.  I'll try and do a little detective work, myself on the values and docs, maybe get a little lucky.

I'll also try the span folks.  I got some good responses from them in DMs, but then they dropped off the HA forums.

**Comment by tylas13** [comment]
My feed-through consumed energy is also negative and I think I know why.  I have my solar inverter system hooked up to the downstream lugs, or "downstream loads" on the Span app, and the wattage closely follows my dual SolarEdge inverter output.  The frustrating thing is the Span app is not set up to do this so it doesn't actually capture my solar power.  Fortunately this integration shows the raw numbers so I can measure it more accurately.

I'll watch at night when the optimizers are using electricity at night to see if it switches over to positive. If I get time I'll set up a db to compare the power output of inverters compared to the downstream output.

**Comment by gdgib** [comment]
Thanks @tylas13. That matches my suspicions, so I wouldn't be surprised. Let me know if I can help in any way.

I do plan to sort this out, but I admit it's not yet the top of my SmartHome development priorities.

**Comment by mbbush** [comment]
There's definitely something goofy about the feedthrough energy numbers reported by the span local api. If I had to bet, I'd say there's a bug somewhere related to signed/unsigned integers and over/underflow, but I haven't been able to find it yet, and it is also not yet on the top of my list of development priorities. 

> What does feed through mean?

These numbers represent the energy transferred through the [feed-through lugs](https://www.se.com/us/en/faqs/FA112516/) attached to the bottom of the panel. These lugs are designed for attaching another panel (smart or otherwise) downstream, and providing up to a 200A connection. If you don't have anything connected to them, you can just ignore these values.

**Comment by gdgib** [comment]
I'm going to leave this out of 0.0.9, where I want to focus on control & connection issues.  0.0.10 is where I'm planning to look at the sensors in more detail.

**Comment by cayossarian** [comment]
Addressed via README.MD - Fixed as of v1.0.x with merges from SpanPanel/SpanCustom.  Maybe there is something else we can do but I don't know of anything.
